spec/rspec/support/object_inspector_spec.rb

```diff
@@ -92,6 +92,46 @@ def with_date_loaded
             matcher_without_a_description.inspect)
         end
       end
+
+      context "with double objects" do
+        it "displays double name" do
+          double = double('Foo')
+          allow(double).to receive(:name).and_return("Foo")
+          ObjectInspector.register double.class do |double|
+            "Double (#{double.name})"
+          end
+          expect(ObjectInspector.inspect(double)).to eq("Double (Foo)")
+        end
+      end
+
+      context "with arbitary type" do
+        class User
+          attr_accessor :name
+
+          def initialize(name)
+            @name = name
+          end
+        end
+
+        it "works with custom class" do
+          _user = User.new("FooBar")
+          ObjectInspector.register User do |user|
+            "User (#{user.name})"
+          end
+          expect(ObjectInspector.inspect(_user)).to eq("User (FooBar)")
+        end
+
+        it "works with subclasses" do
+          class Admin < User
+          end
+
+          _admin = Admin.new("FooBar")
+          ObjectInspector.register Admin do |admin|
+            "Admin (#{admin.name})"
+          end
+          expect(ObjectInspector.inspect(_admin)).to eq("Admin (FooBar)")
+        end
```

Comment:
This spec doesn't actually test what it needs to.  All classes are subclasses (of `Object` if not specified) so it's not actually interesting to register a subclass and test that the registered format logic is used.

What's of interest here is that `Admin`, as a subclass of `User`, should be formatted with the registered `User` logic, w/o the need to register `Admin`.  IOW, we want the format logic to apply to a class and any subclasses, w/o the need to register for each subclass.

This is important, because rspec-mocks has several different types of doubles and we don't want to have to register each subclass to have it formatted properly.